### ISSUE TITLE
Input after ReactEditor.focus doesn't work

### ISSUE THREAD

**Issue by harrisonturton** [ianstormtaylor/slate#3634]
#### What's the current behavior?

![Bug Report](https://user-images.githubusercontent.com/20736299/80164703-0e481500-861d-11ea-8e08-f80dafc07315.gif)

[Sandbox reproducing the issue](https://codesandbox.io/s/slate-05x-4z2yf?file=/src/App.tsx)

The first button directs the focus fine, and I can immediately begin typing. When I click the second button, however, my typing is not caught by the input until I click it again.

The `createEditor()` state in this example is being held by `useState` inside an object. I debated removing this to make it more "minimal", but I think that might be relevant – react perhaps does some weird things behind the scenes with hooks that might be interfering.

#### What's the expected behavior?

I expect to be able to begin typing and have my keypresses caught by the input *immediately* after `ReactEditor.focus` is called, and this should work regardless of how many `createEditor()` instances exist on the page.

**Slate Version**
```
"slate": "^0.57.1",                                                                                                                                                          
"slate-history": "^0.57.1",                                                                                                                                                  
"slate-react": "^0.57.1"                                                                                                                                                     
```

**OS & Browsers I could reproduce the issue on**
- macOS Catalina `10.15.1`
- Firefox `75.0 (64-bit)`
- Chrome `Version 81.0.4044.122 (Official Build) (64-bit)`
- Safari `Version 13.0.3 (15608.3.10.1.4)`

**Comment by CameronAckermanSEL** [comment]
I wasn't able to look at this in depth, but I think this might be caused by the `onBlur` event for the first editor firing after the `onFocus` event of the second editor, and this is causing the focus to be removed from the second editor shortly after it's granted when the button is clicked

**Comment by harrisonturton** [comment]
I thought it might've been because I was storing the editor state inside `useState` instead of `useMemo`, but I tried both and got the same behaviour. I even tried `useReducer`, didn't work either.

**Comment by CameronAckermanSEL** [comment]
Hey @harrisonturton , I know it's been a month but I took a look at this and produced a fork of the original code sandbox.

https://codesandbox.io/s/slate-05x-69xnt

What I've noticed is that this for some reason only seems to happen when clicking `Focus One` then `Focus Two`. Any other combination, even after creating new "pages" seems to work correctly, at least for me. This leads me to think there's something up with how the second editor is being initialized but I don't see anything wrong with it at a glance.

**Comment by tk21** [comment]
I'm coming across the same issue, and the multiple instances of editors are set up much in the same way that is provided by OP's sandbox. What I've tried to do is manually focus on click by calling

`Transforms.select(this.editorRef, this.editorRef?.selection || Editor.end(this.editorRef, []))`

And this actually works _sometimes_ but not all the time when I use the mouse to switch focusing on different editors on the same page. What I'm noticing is that the `editorRef.selection` is sometimes logging as `null`. But I'm not sure why that might be? I tried to take a look at past issues but I couldn't really find anything as relevant as this issue here. 

Also felt like my issue could be related to this closed issue: https://github.com/ianstormtaylor/slate/issues/3338

But I checked my slate-react version and I'm sure that we're using lodash's `throttle` as suggested in that issue ^. Would you happen to have any insight @CameronAckermanSEL or maybe @ianstormtaylor?

@harrisonturton , wondering if you were able to find a suitable workaround for this issue?

**Comment by sebastianseilund** [comment]
I described a workaround in https://github.com/ianstormtaylor/slate/issues/3696#issuecomment-723497830

**TL;DR; Workaround: Make sure that your onChange handler does not cause a re-render if the Slate value has not changed.**

I mad a [working Sandbox fork](https://codesandbox.io/s/slate-05x-forked-id2cw?file=/src/App.tsx) based on @CameronAckermanSEL's non-working example. Make `onChangePage` not change the `pages` state object if the content didn't actually change.

```js
  const onChangePage = (id: number) => (content: Node[]) =>
    setPages((pages) => {
      if (pages[id].content === content) {
        return pages;
      }
      return {
        ...pages,
        [id]: {
          ...pages[id],
          content: content
        }
      };
    });
```
